core/server/api_container/server/service_network/default_service_network.go

```diff
@@ -1725,8 +1769,22 @@ func (network *DefaultServiceNetwork) renderTemplatesUnlocked(templatesAndDataBy
 	if err != nil {
 		return "", stacktrace.Propagate(err, "An error occurred while getting files artifact store")
 	}
-	filesArtifactUuid, err := store.StoreFile(compressedFile, compressedFileMd5, artifactName)
+	existingFilesArtifactUuid, _, existingFilesArtifactMd5, found, err := store.GetFile(artifactName)
 	if err != nil {
+		return "", stacktrace.Propagate(err, "An unexpected error occurred checking for file artifact '%s' existence in the store", artifactName)
+	}
+
+	var storeOrUpdateErr error
+	var filesArtifactUuid enclave_data_directory.FilesArtifactUUID
+	if found {
+		filesArtifactUuid = existingFilesArtifactUuid
+		if len(existingFilesArtifactMd5) == 0 || len(compressedFileMd5) == 0 || !bytes.Equal(existingFilesArtifactMd5, compressedFileMd5) {
```

Comment:
could you add a comment above these conditions or extract them in a helper function with a descriptive name?

Reply:
I can image that you are comparing files content, but I think that a comment could help or a helper method

Reply:
Will do!